Tools/CISupport/ews-build/steps.py

```diff
@@ -728,7 +729,10 @@ def run(self):
             GIT_PASSWORD=access_token,
         )
 
-        return super().run()
+        rc = yield super().run()
+        if rc == SUCCESS and self.getProperty('buildername', '').lower() not in ('commit-queue', 'merge-queue', 'unsafe-merge-queue'):
+            yield self._dovccmd(['remote', 'set-url', '--push', 'origin', 'INVALID'])
```

Comment:
maybe change INVALID to something like PUSH_DISABLED_BY_ADMIN. So it it's easy to debug when someone sees the error message (basically error message will specify why origin is invalid).